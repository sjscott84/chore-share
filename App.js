import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Button
} from 'react-native';
import {
  StackNavigator,
  NavigationActions
} from 'react-navigation';
import * as firebase from 'firebase';

import ListChores from './components/listChores';
import SignIn from './components/signin';
import Family from './components/family';
import AddChore from './components/addChore';
import ChoreDetail from './components/chore';
import EditChore from './components/editChore';

// Initialize Firebase
const firebaseConfig = {
    apiKey: "AIzaSyBshI0DN8DmkDTGiq7_Wj1cBAsYnjCtfkk",
    authDomain: "chore-share.firebaseapp.com",
    databaseURL: "https://chore-share.firebaseio.com",
    projectId: "chore-share",
    storageBucket: "chore-share.appspot.com",
    messagingSenderId: "607979940389"
};

firebase.initializeApp(firebaseConfig);


export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null
    }
  }

  componentWillMount(){
    // Listen for authentication state to change.
    firebase.auth().onAuthStateChanged((user) => {
      if (user != null) {
        console.log("We are authenticated now!");
        this.setState({user: user})
        //this.props.navigation.navigate('ListChores');
      }
    });
  }

  render() {
    if(this.state.user != null){
      return <AuthRoutes />
    }
    return <RoutesConfig />
  }
}

const RoutesConfig = StackNavigator({
  SignIn: {screen: SignIn},
  ListChores: { screen: ListChores},
  Family: {screen: Family},
  AddChore: {screen: AddChore},
  ChoreDetail: {screen: ChoreDetail},
  EditChore: {screen: EditChore}
});

const AuthRoutes = StackNavigator({
  ListChores: { screen: ListChores},
  Family: {screen: Family},
  AddChore: {screen: AddChore},
  ChoreDetail: {screen: ChoreDetail},
  EditChore: {screen: EditChore}
});

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  }
});
