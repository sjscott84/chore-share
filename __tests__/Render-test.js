import 'react-native';
import React from 'react';
import App from '../App';
import SignIn from '../components/signin';
import {
  StackNavigator,
  NavigationActions
} from 'react-navigation';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('App renders correctly', () => {
  const tree = renderer.create(
    <App  />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Sign in renders correctly', () => {
  const tree = renderer.create(
    <SignIn navigation={StackNavigator} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});