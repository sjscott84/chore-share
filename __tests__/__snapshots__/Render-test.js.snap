// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 10,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "rgb(52, 73, 76)",
                "fontFamily": "AppleSDGothicNeo-Medium",
                "fontSize": 24,
                "lineHeight": 33,
                "margin": 10,
                "paddingBottom": 60,
                "paddingTop": 50,
              }
            }
          >
            Login to Chore Share
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "rgb(52, 73, 76)",
                "fontFamily": "AppleSDGothicNeo-SemiBold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 10,
              }
            }
          >
            Email
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "borderColor": "rgb(139, 157, 160)",
                "borderRadius": 2,
                "borderWidth": 1,
                "height": 40,
                "margin": 6,
                "marginBottom": 20,
                "paddingLeft": 5,
              }
            }
            value="s.j.scott84@gmail.com"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "flex-start",
                "color": "rgb(52, 73, 76)",
                "fontFamily": "AppleSDGothicNeo-SemiBold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 10,
              }
            }
          >
            Password
          </Text>
          <TextInput
            onChangeText={[Function]}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "stretch",
                "borderColor": "rgb(139, 157, 160)",
                "borderRadius": 2,
                "borderWidth": 1,
                "height": 40,
                "margin": 6,
                "marginBottom": 20,
                "paddingLeft": 5,
              }
            }
            value="C228kp3jE"
          />
          <View
            accessibilityComponentType="button"
            accessibilityLabel={undefined}
            accessibilityTraits={
              Array [
                "button",
              ]
            }
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#0C42FD",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Sign In
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      cardStyle={undefined}
      collapsable={undefined}
      getScreenDetails={[Function]}
      headerMode={undefined}
      index={0}
      layout={
        Object {
          "height": 0,
          "initHeight": 0,
          "initWidth": 0,
          "isMeasured": false,
          "width": 0,
        }
      }
      mode="float"
      navigation={
        Object {
          "dispatch": [Function],
          "goBack": [Function],
          "navigate": [Function],
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "routes": Array [
              Object {
                "key": "Init-id-1502771829064-0",
                "routeName": "SignIn",
              },
            ],
          },
        }
      }
      router={
        Object {
          "getActionForPathAndParams": [Function],
          "getComponentForRouteName": [Function],
          "getComponentForState": [Function],
          "getPathAndParamsForState": [Function],
          "getScreenConfig": [Function],
          "getScreenOptions": [Function],
          "getStateForAction": [Function],
        }
      }
      style={
        Object {
          "backgroundColor": "#EFEFF2",
          "elevation": 4,
          "height": 64,
          "paddingTop": 20,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0.5,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
        }
      }
      transitionConfig={undefined}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "justifyContent": "center",
                "left": 70,
                "opacity": 1,
                "position": "absolute",
                "right": 70,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <Text
              accessibilityTraits="header"
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, .9)",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "marginHorizontal": 16,
                    "textAlign": "center",
                  },
                  Array [
                    null,
                    undefined,
                  ],
                ]
              }
            >
              Welcome
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Sign in renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "rgb(52, 73, 76)",
        "fontFamily": "AppleSDGothicNeo-Medium",
        "fontSize": 24,
        "lineHeight": 33,
        "margin": 10,
        "paddingBottom": 60,
        "paddingTop": 50,
      }
    }
  >
    Login to Chore Share
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "rgb(52, 73, 76)",
        "fontFamily": "AppleSDGothicNeo-SemiBold",
        "fontSize": 15,
        "lineHeight": 20,
        "marginLeft": 10,
      }
    }
  >
    Email
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "borderColor": "rgb(139, 157, 160)",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 40,
        "margin": 6,
        "marginBottom": 20,
        "paddingLeft": 5,
      }
    }
    value="s.j.scott84@gmail.com"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "rgb(52, 73, 76)",
        "fontFamily": "AppleSDGothicNeo-SemiBold",
        "fontSize": 15,
        "lineHeight": 20,
        "marginLeft": 10,
      }
    }
  >
    Password
  </Text>
  <TextInput
    onChangeText={[Function]}
    secureTextEntry={true}
    style={
      Object {
        "alignSelf": "stretch",
        "borderColor": "rgb(139, 157, 160)",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 40,
        "margin": 6,
        "marginBottom": 20,
        "paddingLeft": 5,
      }
    }
    value="C228kp3jE"
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
